ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -106,34 +122,40 @@ sub Configure_Cache {
         $self->warn("Cannot determine current AFS cache size, changing only config file");
     }
 
-    my $afs_cacheinfo_fh = CAF::FileEditor->new( $AFS_CACHEINFO, log => $self ); # TODO use FileReader
-    $afs_cacheinfo_fh->cancel();
-    if ( "$afs_cacheinfo_fh" =~ m;^([^:]+):([^:]+):(\d+)$; ) {
-        $file_afsmount   = $1;
-        $file_cachemount = $2;
+    my $afs_cacheinfo_fh = CAF::FileReader->new( $AFS_CACHEINFO, log => $self );
+    $self->debug(2, "$AFS_CACHEINFO current contents: >>>$afs_cacheinfo_fh<<<");
+    if ( "$afs_cacheinfo_fh" =~ m;^([^:]+):([^:]+):(\d+|AUTOMATIC)$; ) {
+        $file_afsmount   = $1 if $file_afsmount eq "";
+        $file_cachemount = $2 if $file_cachemount eq "";
         $file_cache      = $3;
-    }
-    else {
+    } elsif ( "$afs_cacheinfo_fh" ne "" ) {
         $self->error("Cannot parse stored AFS cache mount or size from $AFS_CACHEINFO");
```

Comment:
Not sure it adds anything... It is enough to look at fhe `cacheinfo` file contents.